datafusion/sqllogictest/bin/sqllogictests.rs

```diff
@@ -39,6 +40,23 @@ pub fn main() -> Result<()> {
         .block_on(run_tests())
 }
 
+fn value_validator(actual: &[Vec<String>], expected: &[String]) -> bool {
+    let expected = expected
+        .iter()
+        // Trailing whitespace from lines in SLT will typically be removed, but do not fail if it is not
+        // If particular test wants to cover trailing whitespace on a value,
+        // it should project additional non-whitespace column on the right.
+        .map(|s| s.trim_end().to_owned())
+        .collect::<Vec<_>>();
+    let actual = actual
+        .iter()
+        .map(|strs| strs.iter().join(" "))
```

Comment:
we prob can do this in a sinlge map?

Reply:
I am not quite sure what you are suggesting here

Reply:
I'm suggesting to have a single collection iteration instead of two
```
.map(|strs| strs.iter().join(" ").trim_end().to_owned())
```

Reply:
yes, we can do this in single map.
i placed this into two map() steps so that the common part of actual and expected processing _looks the same_ and so that i can write a nice comment